### PR TITLE
Expose instruction builders for Openbook

### PR DESCRIPTION
**Pull request by mschneider**
* create OO
* place order
* cancel all

also removed unneeded result/error return type that was technically dead code

### CODE REVIEW

File: lib/client/src/client.rs [hunk 1]
```diff
@@ -429,36 +429,66 @@ impl MangoClient {
...
+        let quote_info = self.context.token(serum3_info.market.quote_token_index);
+        let base_info = self.context.token(serum3_info.market.base_token_index);
+
+        Serum3Data {
+            market_index,
+            market: serum3_info,
+            quote: quote_info,
+            base: base_info,
+        }
+    }
```
[the remark this answers is not in the document]
Comment (mschneider) [reply lib/client/src/client.rs]: yeah moved them over, looks a little bit better now

File: bin/keeper/src/taker.rs [hunk 2]
```diff
@@ -157,32 +157,42 @@ pub async fn loop_blocking_orders(
...
-                    anyhow::bail!("Price RwLock PoisonError!");
-                }
-            };
+            let fresh_price = price.read().unwrap().to_num::<f64>();
+            let bid_price = fresh_price * 1.1;
 
-            let fresh_price = fresh_price.to_num::<f64>();
+            let bid_price_lots =
+                bid_price * 10f64.powi(quote_decimals - base_decimals) * s3.coin_lot_size as f64
+                    / s3.pc_lot_size as f64;
```
Comment (ckamm) [inline bin/keeper/src/taker.rs:187]: might be nice to have price_ui_to_serum_lots() or something (I'm unclear on whether `fresh_price` is the ui price or native price or something else)

File: lib/client/src/client.rs [hunk 3]
```diff
@@ -515,15 +481,21 @@ impl MangoClient {
         account: &MangoAccountValue,
         market_index: Serum3MarketIndex,
         side: Serum3Side,
-        price: f64,
-        size: f64,
+        limit_price: u64,
+        max_base_qty: u64,
+        max_native_quote_qty_including_fees: u64,
```
Comment (ckamm) [inline lib/client/src/client.rs:486]: There definitely should be a simple ix constructor like this one; but wondering if we should keep mc's idea of having a simplified one with ui numbers. 

I guess this client is for advanced users who're ok figuring out "quote_qty_incl_fees" themselves. 

I'd like if it was `limit_price_lots` and `max_base_qty_lots` to be clearer about the units of these two.

File: lib/client/src/client.rs [hunk 4]
```diff
@@ -532,65 +504,9 @@ impl MangoClient {
...
-            (price * ((10u64.pow(s3.quote.decimals as u32) * s3.market.coin_lot_size) as f64))
-                as u64
-                / (10u64.pow(s3.base.decimals as u32) * s3.market.pc_lot_size)
-        };
-        // https://github.com/project-serum/serum-ts/blob/master/packages/serum/src/market.ts#L1333
-        let max_base_qty =
-            { (size * 10u64.pow(s3.base.decimals as u32) as f64) as u64 / s3.market.coin_lot_size };
-        let max_native_quote_qty_including_fees = {
-            fn get_fee_tier(msrm_balance: u64, srm_balance: u64) -> u64 {
-                if msrm_balance >= 1 {
```
Comment (ckamm) [inline lib/client/src/client.rs:535]: oh yeah, this was very very out of date

File: bin/keeper/src/taker.rs (hunk 2 above)
Comment (mschneider) [reply bin/keeper/src/taker.rs:187]: openbook v2 sdk is going to have that imo

File: lib/client/src/client.rs (hunk 3 above)
Comment (mschneider) [reply lib/client/src/client.rs:486]: ui numbers should be another layer on top, ideally a rest api

**Pull request merged**